jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/Dispatcher.java

```diff
@@ -153,38 +148,29 @@ public void async(ServletRequest request, ServletResponse response) throws Servl
 
     public class ParameterRequestWrapper extends HttpServletRequestWrapper
     {
-        private final MultiMap<String> _params = new MultiMap<>();
-        private boolean decodedParams = false;
-        private final ServletContextRequest _request;
+        private MultiMap<String> _params = null;
 
         public ParameterRequestWrapper(HttpServletRequest request)
         {
             super(request);
-
-            // Have to assume ENCODING because we can't know otherwise.
-            String targetQuery = (_uri == null) ? null : _uri.getQuery();
-            if (targetQuery != null)
-                UrlEncoded.decodeTo(targetQuery, _params, UrlEncoded.ENCODING);
-
-            _request = ServletContextRequest.getServletContextRequest(request);
-
-            Fields queryParams = _request.getServletApiRequest().getQueryParams();
-            for (Fields.Field field : queryParams)
-            {
-                _params.addValues(field.getName(), field.getValues());
-            }
         }
 
         private MultiMap<String> getParams()
         {
-            if (decodedParams)
-                return _params;
-            decodedParams = true;
-
-            Fields contentParams = _request.getServletApiRequest().getContentParams();
-            for (Fields.Field field : contentParams)
+            if (_params == null)
```

Comment:
Because if the target of the dispatch never calls any parameter methods, then there is no need to build the merged params.  In fact there is an optimisation available if the target has no query parameters... but leaving that for now.